libopenage/util/compiler.cpp

```diff
@@ -39,12 +45,73 @@ std::string addr_to_string(const void *addr) {
 	return sformat("[%p]", addr);
 }
 
+#ifdef _MSC_VER
+namespace
+{
+
+struct symbol_name_result
+{
+	// Better ignore the result if the initialization has failed.
+	bool initialization_fail = false;
+
+	std::optional<std::string> symbol_name = {};
+};
+
+symbol_name_result symbol_name_win(const void *addr)
+{
+	static HANDLE process_handle = INVALID_HANDLE_VALUE;
+	static bool initialized_symbol_handler = false;
+	static bool initialized_symbol_handler_successfully = false;
+
+	// Initialize symbol handler for process, if it has not yet been initialized
+	// If we are not succesful on the first try, leave it, since MSDN says that searching for symbol files is very time consuming
+	if (!initialized_symbol_handler)
+	{
+		initialized_symbol_handler = true;
+
+		process_handle = GetCurrentProcess();
+		initialized_symbol_handler_successfully = SymInitialize(process_handle, nullptr, TRUE);
+	}
+
+	symbol_name_result result = {};
+	if (!initialized_symbol_handler_successfully)
+	{
+		result.initialization_fail = true;
+		return result;
+	}
+
+	// The magic of winapi
+	constexpr int name_buffer_size = 1024;
+	char buffer[sizeof(SYMBOL_INFO) + name_buffer_size * sizeof(char)];
```

Comment:
Hey. Yes it is true that sizeof(char) is defined as 1. Here is my intention on doing so: If I left out the sizeof part, imo it would not have been 100% clear why I am adding the buffer size to the buffer. I would have left a comment like "Allocate a buffer that can hold the size of the struct + a string of name_buffer_size". But by adding the sizeof(char), I think that it becomes immediatly clear that the thing I'm adding is meant to be a string of that size. Of course I can still change it. :)